.github/workflows/openssl_versions.yml

```diff
@@ -0,0 +1,153 @@
+---
+name: OpenSSL Versions
+
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    name: Versions
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: ossl30
+            build: true
+            version: "3.0"
+            container: fedora:latest
+          - name: ossl32
+            build: true
+            version: "3.2"
+            container: fedora:latest
+          - name: ossl35
+            build: true
+            version: "3.5"
+            container: fedora:latest
+          - name: centos10
+            build: false
+            version: "3.5"
+            container: quay.io/centos/centos:stream10
+    container: ${{ matrix.container }}
+    steps:
+      - name: Get Date for DNF cache entry
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%V")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Restore DNF cache
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            /var/cache/libdnf5
+          key: ${{ runner.os }}-dnf-${{ steps.get-date.outputs.date }}
+
+      - name: Install Dependencies
+        run: |
+          if [ "${{ matrix.name }}" = "centos" ]; then
+           dnf_opts="--enablerepo=crb"
+          fi
+          if [ -f /etc/redhat-release ]; then
+            dnf -y install git cargo clang-devel \
+              'perl(FindBin)' 'perl(lib)' 'perl(File::Compare)' \
+              'perl(File::Copy)' 'perl(bigint)' 'perl(Time::HiRes)' \
+              'perl(IPC::Cmd)' 'perl(Pod::Html)' 'perl(Digest::SHA)' \
+              'perl(Module::Load::Conditional)' 'perl(File::Temp)' \
+              'perl(Test::Harness)' 'perl(Test::More)' 'perl(Math::BigInt)' \
+              zlib-devel sed sqlite-devel
+            if [ "${{ matrix.build }}" = "false" ]; then
+              dnf -y install openssl-devel
+            fi
+          elif [ -f /etc/debian_version ]; then
+            apt-get -q update
+            apt-get -yq install git clang curl
+            if [ "${{ matrix.build }}" = "false" ]; then
+              apt-get -yq install libssl-dev
+            fi
+          fi
```

Comment:
if the debian is out of the picture, I would remove these parts to simplify the script.  Same for the caching of rust dependencies, where the debian is also checked.